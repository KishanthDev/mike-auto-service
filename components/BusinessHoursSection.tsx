"use client";

import type { BusinessHour } from "@/types/data";

import React from "react";

interface BusinessSectionProps {
  businessHours: BusinessHour[];
}
export default function BusinessHoursSection({
  businessHours,
}: BusinessSectionProps) {
  return (
    <section>
      <h2 className="text-2xl md:text-3xl font-bold border-b-2 border-red-600 pb-2 text-[#1a3c6e] dark:text-white">
        Business Hours
      </h2>

      <div className="mt-4 md:mt-6 space-y-2 text-sm md:text-base text-gray-700 dark:text-gray-300">
        {businessHours.map((day, idx) => (
          <div
            key={idx}
            className="flex justify-between border-b py-2 border-gray-200 dark:border-gray-600"
          >
            <span>{day.day}</span>
            <span>{day.hours}</span>
          </div>
        ))}
        <p className="mt-3 font-semibold text-red-600 dark:text-red-400">
          24/7 Emergency Towing Available
        </p>
      </div>
    </section>
  );
}
