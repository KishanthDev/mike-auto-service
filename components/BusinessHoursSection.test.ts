import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import BusinessHoursSection from './BusinessHoursSection';
import type { BusinessHour } from '@/types/data';

describe('BusinessHoursSection', () => {
  const mockBusinessHours: BusinessHour[] = [
    { day: 'Monday', hours: '8:00 AM - 6:00 PM' },
    { day: 'Tuesday', hours: '8:00 AM - 6:00 PM' },
    { day: 'Wednesday', hours: '8:00 AM - 6:00 PM' },
    { day: 'Thursday', hours: '8:00 AM - 8:00 PM' },
    { day: 'Friday', hours: '8:00 AM - 6:00 PM' },
    { day: 'Saturday', hours: '9:00 AM - 4:00 PM' },
    { day: 'Sunday', hours: 'Closed' },
  ];

  it('renders the section with correct heading', () => {
    render(React.createElement(BusinessHoursSection, { businessHours: mockBusinessHours }));
    
    const heading = screen.getByRole('heading', { level: 2 });
    expect(heading).toBeInTheDocument();
    expect(heading).toHaveTextContent('Business Hours');
    expect(heading).toHaveClass('text-2xl', 'md:text-3xl', 'font-bold', 'border-b-2', 'border-red-600', 'pb-2', 'text-[#1a3c6e]', 'dark:text-white');
  });

  it('renders the emergency towing notice', () => {
    render(React.createElement(BusinessHoursSection, { businessHours: mockBusinessHours }));
    
    const notice = screen.getByText('24/7 Emergency Towing Available');
    expect(notice).toBeInTheDocument();
    expect(notice).toHaveClass('mt-3', 'font-semibold', 'text-red-600', 'dark:text-red-400');
  });

  it('handles empty business hours array', () => {
    render(React.createElement(BusinessHoursSection, { businessHours: [] }));
    
    expect(screen.getByText('Business Hours')).toBeInTheDocument();
    expect(screen.getByText('24/7 Emergency Towing Available')).toBeInTheDocument();
    
    mockBusinessHours.forEach(day => {
      expect(screen.queryByText(day.day)).not.toBeInTheDocument();
      expect(screen.queryByText(day.hours)).not.toBeInTheDocument();
    });
  });

  it('matches snapshot with data', () => {
    const { asFragment } = render(
      React.createElement(BusinessHoursSection, { businessHours: mockBusinessHours })
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('matches snapshot with empty data', () => {
    const { asFragment } = render(
      React.createElement(BusinessHoursSection, { businessHours: [] })
    );
    expect(asFragment()).toMatchSnapshot();
  });
});