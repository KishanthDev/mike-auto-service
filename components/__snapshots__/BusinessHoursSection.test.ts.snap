// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BusinessHoursSection matches snapshot with data 1`] = `
<DocumentFragment>
  <section>
    <h2
      class="text-2xl md:text-3xl font-bold border-b-2 border-red-600 pb-2 text-[#1a3c6e] dark:text-white"
    >
      Business Hours
    </h2>
    <div
      class="mt-4 md:mt-6 space-y-2 text-sm md:text-base text-gray-700 dark:text-gray-300"
    >
      <div
        class="flex justify-between border-b py-2 border-gray-200 dark:border-gray-600"
      >
        <span>
          Monday
        </span>
        <span>
          8:00 AM - 6:00 PM
        </span>
      </div>
      <div
        class="flex justify-between border-b py-2 border-gray-200 dark:border-gray-600"
      >
        <span>
          Tuesday
        </span>
        <span>
          8:00 AM - 6:00 PM
        </span>
      </div>
      <div
        class="flex justify-between border-b py-2 border-gray-200 dark:border-gray-600"
      >
        <span>
          Wednesday
        </span>
        <span>
          8:00 AM - 6:00 PM
        </span>
      </div>
      <div
        class="flex justify-between border-b py-2 border-gray-200 dark:border-gray-600"
      >
        <span>
          Thursday
        </span>
        <span>
          8:00 AM - 8:00 PM
        </span>
      </div>
      <div
        class="flex justify-between border-b py-2 border-gray-200 dark:border-gray-600"
      >
        <span>
          Friday
        </span>
        <span>
          8:00 AM - 6:00 PM
        </span>
      </div>
      <div
        class="flex justify-between border-b py-2 border-gray-200 dark:border-gray-600"
      >
        <span>
          Saturday
        </span>
        <span>
          9:00 AM - 4:00 PM
        </span>
      </div>
      <div
        class="flex justify-between border-b py-2 border-gray-200 dark:border-gray-600"
      >
        <span>
          Sunday
        </span>
        <span>
          Closed
        </span>
      </div>
      <p
        class="mt-3 font-semibold text-red-600 dark:text-red-400"
      >
        24/7 Emergency Towing Available
      </p>
    </div>
  </section>
</DocumentFragment>
`;

exports[`BusinessHoursSection matches snapshot with empty data 1`] = `
<DocumentFragment>
  <section>
    <h2
      class="text-2xl md:text-3xl font-bold border-b-2 border-red-600 pb-2 text-[#1a3c6e] dark:text-white"
    >
      Business Hours
    </h2>
    <div
      class="mt-4 md:mt-6 space-y-2 text-sm md:text-base text-gray-700 dark:text-gray-300"
    >
      <p
        class="mt-3 font-semibold text-red-600 dark:text-red-400"
      >
        24/7 Emergency Towing Available
      </p>
    </div>
  </section>
</DocumentFragment>
`;
